### PR TITLE
[dashboard] github app installation

### PR DESCRIPTION
**Pull request by svenefftinge**
## How to Test

 - Go to https://github.com/apps/gitpod-dev-app
 - Click configure
 - go through the flow
 - verify that prebuilds are triggered when pushing to repos you configured the app on.

### CODE REVIEW

**Comment by svenefftinge** [comment]
### TODO: Remove changes in `values.dev.yaml` before merge

**Comment by svenefftinge** [comment]
fixes #https://github.com/gitpod-io/gitpod/issues/3702

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -1205,7 +1205,7 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         const user = this.checkAndBlockUser();
 
         if (!this.env.githubAppEnabled) {
-            throw new ResponseError(ErrorCodes.NOT_FOUND, 'User is not authenticated. Please login.');
+            throw new ResponseError(ErrorCodes.NOT_FOUND, 'No GitHub enabled for this installation. Please talk to your administrator.');
```
Comment (svenefftinge) [inline components/server/src/workspace/gitpod-server-impl.ts:1208]: Should be 'No GitHub app enabled ....'

**Review by csweichel — APPROVED** [review]
LGTM

File: components/dashboard/src/prebuilds/InstallGitHubApp.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            <div className="px-6 py-3 flex justify-between space-x-2 text-gray-400">
+                <div className="flex flex-col items-center m-auto max-w-lg mt-40">
+                    <h3 className="text-center pb-3 text-gray-500">Install GitHub App</h3>
+                    <div className="text-center pb-6 text-gray-500">You are about to install the GitHub app for Gitpod.</div>
+                    <div className="flex rounded-md bg-gray-200 p-3">
+                        <img className="w-4 h-4 ml-2 mr-3 mt-1" src={info} alt="info" />
+                        <span className="text-gray-500">This action will also allow Gitpod to access private repositories. You can edit git provider permissions later in user settings.</span>
+                    </div>
+                    <div className="mt-6">
+                        <button className="secondary">Cancel</button>
```
Comment (gtsiolis) [inline components/dashboard/src/prebuilds/InstallGitHubApp.tsx:77]: **issue**: This went through without no action. We could potentially remove the button all together. /cc @csweichel @svenefftinge

**Pull request merged**